src/Sentry/Internal/SentryScopeManager.cs

```diff
@@ -101,6 +101,33 @@ public void WithScope(Action<Scope> scopeCallback)
         }
     }
 
+    public T? WithScope<T>(Func<Scope, T?> scopeCallback)
+    {
+        using (PushScope())
+        {
+            var scope = GetCurrent();
+            return scopeCallback.Invoke(scope.Key);
+        }
+    }
+
+    public async Task WithScopeAsync(Func<Scope, Task> scopeCallback)
+    {
+        using (PushScope())
+        {
+            var scope = GetCurrent();
+            await scopeCallback.Invoke(scope.Key).ConfigureAwait(false);
```

Comment:
Do we want to use something like [`ConfigureAwaitChecker`](https://github.com/cincuranet/ConfigureAwaitChecker) analyzer for development to avoid missing `ConfigureAwait()`?

Reply:
See #2308.  Thanks.